test/cpp/microbenchmarks/bm_filters.cc

```diff
@@ -0,0 +1,200 @@
+
+typedef FilterFixture<&grpc_client_channel_filter, 0> ClientChannelFilter;
+typedef FilterBM<ClientChannelFilter> ClientChannelFilterBM;
+BENCHMARK_TEMPLATE(BM_CallStackInit, ClientChannelFilterBM);
+
+typedef FilterFixture<&grpc_message_compress_filter, CHECKS_NOT_LAST>
+    CompressFilter;
+typedef FilterBM<CompressFilter> CompressFilterBM;
+BENCHMARK_TEMPLATE(BM_CallStackInit, CompressFilterBM);
+
+typedef FilterFixture<&grpc_client_deadline_filter, CHECKS_NOT_LAST>
+    ClientDeadlineFilter;
+typedef FilterBM<ClientDeadlineFilter> ClientDeadlineFilterBM;
+BENCHMARK_TEMPLATE(BM_CallStackInit, ClientDeadlineFilterBM);
+
+typedef FilterFixture<&grpc_server_deadline_filter, CHECKS_NOT_LAST>
+    ServerDeadlineFilter;
+typedef FilterBM<ServerDeadlineFilter> ServerDeadlineFilterBM;
+BENCHMARK_TEMPLATE(BM_CallStackInit, ServerDeadlineFilterBM);
+
+typedef FilterFixture<&grpc_http_client_filter,
+                      CHECKS_NOT_LAST | REQUIRES_TRANSPORT>
+    HttpClientFilter;
+typedef FilterBM<HttpClientFilter> HttpClientFilterBM;
+BENCHMARK_TEMPLATE(BM_CallStackInit, HttpClientFilterBM);
+
+typedef FilterFixture<&grpc_http_server_filter, CHECKS_NOT_LAST>
+    HttpServerFilter;
+typedef FilterBM<HttpServerFilter> HttpServerFilterBM;
+BENCHMARK_TEMPLATE(BM_CallStackInit, HttpServerFilterBM);
+
+typedef FilterFixture<&grpc_message_size_filter, CHECKS_NOT_LAST>
+    MessageSizeFilter;
+typedef FilterBM<MessageSizeFilter> MessageSizeFilterBM;
+BENCHMARK_TEMPLATE(BM_CallStackInit, MessageSizeFilterBM);
+
+typedef FilterFixture<&grpc_server_load_reporting_filter, CHECKS_NOT_LAST>
+    ServerLoadReportingFilter;
+typedef FilterBM<ServerLoadReportingFilter> ServerLoadReportingFilterBM;
+BENCHMARK_TEMPLATE(BM_CallStackInit, ServerLoadReportingFilterBM);
+
+// Measure full filter functionality overhead, from initializing the call stack
+// through running all filter callbacks.
+// Note that all we do is send down all 6 ops through the filter stack; we do
+// not test different combinations or subsets of ops. Thus, this test does not
+// comprehensively test all the code paths of each individual filter because
+// filters may take different code paths based on the combination and/or
+// ordering of the ops.
+template <class FilterBM>
+static void BM_FullFilterFunctionality(benchmark::State& state) {
+  // Setup for benchmark
+  FilterBM bm_setup;
+  struct DataForFilterBM data;
+  bm_setup.Setup(&data);
+
+  // Run the benchmark
+  while (state.KeepRunning()) {
+    GPR_TIMER_SCOPE("BenchmarkCycle", 0);
+    // Because it's not valid to send more than one of any of the {send, recv}_
+    // {initial, trailing}_metadata ops on a single call, we need to construct
+    // a new call stack each time through the loop. It's also not valid to have
+    // more than one of send_message or recv_message in flight on a single call
+    // at the same time.
+    memset(data.call_stack, 0, data.channel_stack->call_stack_size);
```

Comment:
Another interesting data point for streaming would be pulling this out of the main loop, then only benchmarking batches with send or recv message batches.

Is that the plan for this continuation of this project? I know you mentioned you had more coming.

Reply:
Sounds good